### PR TITLE
Move checkAndUpdateWireless into VerifyPending and also handle override

### PR DESCRIPTION
**Pull request by naiming-zededa**
Signed-off-by: Naiming Shen <naiming@zededa.com>
- moved the checkAndUpdateWireless() into VerifyPending()
- leave the 'override' type of update also at the doUpdatePortConfigListAndPublish() to catch the case when the wifi override might not get into later verify stage
- change the hack of ignore zedcloud wifi null configure by copying the override wifi config into other types of update, thus now the lastresort config will have the wificonfig set by override
- tested on my device with baseimage upgrade, reboot, reboot with only wifi available

### CODE REVIEW

**Comment by rvs** [comment]
This looks reasonable @naiming-zededa -- but I guess we still need to address the issue of timing between this portion of nim and other device-steps sections (this is only relevant when time is completely out of whack on boot).

I suppose we can do it in a separate PR, but it would be great to address it (unless I somehow missed how this PR addresses it).

File: pkg/pillar/devicenetwork/dnc.go [hunk 1]
```diff
@@ -651,6 +655,11 @@ func (ctx *DeviceNetworkContext) doUpdatePortConfigListAndPublish(
 	current := getCurrentDPC(ctx) // Used to determine if index needs to change
 	currentIndex := ctx.DevicePortConfigList.CurrentIndex
 	oldConfig, _ := lookupPortConfig(ctx, *portConfig)
+	if strings.Contains(portConfig.Key, "override") {
```
Comment (eriknordmark) [inline pkg/pillar/devicenetwork/dnc.go]: Can you please put a // XXX remove this code when the controller supports WiFi config
comment so we can find it and zap it later?

File: pkg/pillar/devicenetwork/dnc.go [hunk 2]
```diff
@@ -734,18 +736,7 @@ func checkAndUpdateWireless(ctx *DeviceNetworkContext, oCfg *types.DevicePortCon
...
-			// has wifi information, but the new update does not. Remove this when the zedcloud
-			// implements the wifi configure for port
-			if oldPortCfg != nil && oldPortCfg.WirelessCfg.WType == types.WirelessTypeWifi &&
-				pCfg.WirelessCfg.WType == types.WirelessTypeNone {
-				log.Infof("checkAndUpdateWireless: HACK! wifi has configure %+v, ignore the update %+v\n",
-					oldPortCfg.WirelessCfg, pCfg.WirelessCfg)
-				break
-			}
-			// XXX
+		if oldPortCfg == nil || isOverride || !reflect.DeepEqual(oldPortCfg.WirelessCfg, pCfg.WirelessCfg) {
```
Comment (eriknordmark) [inline pkg/pillar/devicenetwork/dnc.go]: Once controller supports WiFi config, shouldn't the 'isOverride' check go away here as well? Or the whole comparison of WirelessCfg should go away? 
Please add comment indicating what should be removed once we have controller support.

File: pkg/pillar/devicenetwork/dnc.go [hunk 3]
```diff
@@ -755,6 +746,25 @@ func checkAndUpdateWireless(ctx *DeviceNetworkContext, oCfg *types.DevicePortCon
 				log.Infof("checkAndUpdateWireless: updated wpa file ok %v\n", status)
 			}
 		}
+
+		// hack, save the override wifi param into ctx for later copying
```
Comment (eriknordmark) [inline pkg/pillar/devicenetwork/dnc.go]: // XXX remove when controller support WiFi
would make it easier to find, especially if you use the same comment string in all places where removal is needed.

**Comment by naiming-zededa** [comment]
@eriknordmark addressed the comment on proper comments when controller supports this. Will do another testing/debug/verify later when controller is ready. There might need to be some change by then.

**Comment by eriknordmark** [comment]
I only see the XXX addition in one place. Did you intend to add it where we need to delete code as well?

**Comment by naiming-zededa** [comment]
Hmm, not sure what happened. my local diff
git diff fdb8ae6f9b3ac48359917c65318ab8efcad5b224 acf0bb7f50d4c10b4525d9874e3fb6beb74e25f1


showing XXX in 3 places…

**Comment by naiming-zededa** [comment]
move this PR to #521

**Comment by naiming-zededa** [comment]
move this PR to #521

**Pull request closed without merging**